spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -373,4 +386,371 @@ public void testNotIn() {
+    Expression actualLt2 = SparkV2Filters.convert(lt2);
+    Assertions.assertThat(actualLt2.toString()).isEqualTo(expectedLt2.toString());
+
+    Predicate ltEq1 = new Predicate("<=", attrAndValue);
+    Expression expectedLtEq1 = Expressions.lessThanOrEqual(item, value);
+    Expression actualLtEq1 = SparkV2Filters.convert(ltEq1);
+    Assertions.assertThat(actualLtEq1.toString()).isEqualTo(expectedLtEq1.toString());
+
+    Predicate ltEq2 = new Predicate("<=", valueAndAttr);
+    Expression expectedLtEq2 = Expressions.greaterThanOrEqual(item, value);
+    Expression actualLtEq2 = SparkV2Filters.convert(ltEq2);
+    Assertions.assertThat(actualLtEq2.toString()).isEqualTo(expectedLtEq2.toString());
+
+    Predicate gt1 = new Predicate(">", attrAndValue);
+    Expression expectedGt1 = Expressions.greaterThan(item, value);
+    Expression actualGt1 = SparkV2Filters.convert(gt1);
+    Assertions.assertThat(actualGt1.toString()).isEqualTo(expectedGt1.toString());
+
+    Predicate gt2 = new Predicate(">", valueAndAttr);
+    Expression expectedGt2 = Expressions.lessThan(item, value);
+    Expression actualGt2 = SparkV2Filters.convert(gt2);
+    Assertions.assertThat(actualGt2.toString()).isEqualTo(expectedGt2.toString());
+
+    Predicate gtEq1 = new Predicate(">=", attrAndValue);
+    Expression expectedGtEq1 = Expressions.greaterThanOrEqual(item, value);
+    Expression actualGtEq1 = SparkV2Filters.convert(gtEq1);
+    Assertions.assertThat(actualGtEq1.toString()).isEqualTo(expectedGtEq1.toString());
+
+    Predicate gtEq2 = new Predicate(">=", valueAndAttr);
+    Expression expectedGtEq2 = Expressions.lessThanOrEqual(item, value);
+    Expression actualGtEq2 = SparkV2Filters.convert(gtEq2);
+    Assertions.assertThat(actualGtEq2.toString()).isEqualTo(expectedGtEq2.toString());
+
+    Predicate eq1 = new Predicate("=", attrAndValue);
+    Expression expectedEq1 = Expressions.equal(item, value);
+    Expression actualEq1 = SparkV2Filters.convert(eq1);
+    Assertions.assertThat(actualEq1.toString()).isEqualTo(expectedEq1.toString());
+
+    Predicate eq2 = new Predicate("=", valueAndAttr);
+    Expression expectedEq2 = Expressions.equal(item, value);
+    Expression actualEq2 = SparkV2Filters.convert(eq2);
+    Assertions.assertThat(actualEq2.toString()).isEqualTo(expectedEq2.toString());
+
+    Predicate notEq1 = new Predicate("<>", attrAndValue);
+    Expression expectedNotEq1 = Expressions.notEqual(item, value);
+    Expression actualNotEq1 = SparkV2Filters.convert(notEq1);
+    Assertions.assertThat(actualNotEq1.toString()).isEqualTo(expectedNotEq1.toString());
+
+    Predicate notEq2 = new Predicate("<>", valueAndAttr);
+    Expression expectedNotEq2 = Expressions.notEqual(item, value);
+    Expression actualNotEq2 = SparkV2Filters.convert(notEq2);
+    Assertions.assertThat(actualNotEq2.toString()).isEqualTo(expectedNotEq2.toString());
+
+    Predicate eqNullSafe1 = new Predicate("<=>", attrAndValue);
```

Comment:
I think this needs a couple more cases for equality and null-safe equality where the value is null. Null-safe equality should convert a null value to `isNull` and equality should not convert the expression (returns null).

Reply:
Previously, added comparisons between reference and `NULL` here https://github.com/apache/iceberg/blob/c27c1fb1f279c8be96847fc41ce637b3eff19661/spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java#L207

Should we still extend to UDF as well?